xmtp/src/vmac.rs

```diff
@@ -0,0 +1,54 @@
+use xmtp_proto::xmtp::v3::message_contents::{
+    VmacAccountLinkedKey, VmacContactBundle, VmacDeviceLinkedKey, VmacUnsignedPublicKey,
+};
+
+use serde_json::json;
+use vodozemac::{
+    olm::{Account, SessionConfig},
+};
+
+use crate::vmac_traits::ProtoWrapper;
+
+// Generate a VmacContactBundle
+pub fn generate_test_contact_bundle() -> VmacContactBundle {
+    let mut account = Account::new();
+    // Get account identity key
+    let identity_key = account.curve25519_key();
+    account.generate_fallback_key();
+    let fallback_key = account.fallback_key().values().next().unwrap().to_owned();
+
+    let identity_key_proto: ProtoWrapper<VmacUnsignedPublicKey> = identity_key.into();
+    let fallback_key_proto: ProtoWrapper<VmacUnsignedPublicKey> = fallback_key.into();
+    let identity_key = VmacAccountLinkedKey {
+        key: Some(identity_key_proto.proto),
+    };
+    let fallback_key = VmacDeviceLinkedKey {
+        key: Some(fallback_key_proto.proto),
+    };
+    VmacContactBundle {
+        identity_key: Some(identity_key),
+        prekey: Some(fallback_key),
+    }
+}
+
+// Generate an outbound session (Olm Prekey Message) given a VmacContactBundle
+pub fn generate_outbound_session(bundle: VmacContactBundle) -> Vec<u8> {
+    let account = Account::new();
```

Comment:
How is this function intended to function? 

Is an account generated for every session?

Reply:
I should prefix with `test_` - meant almost as a demo that the proto code isn't just baloney

Reply:
actually moved all of the functions directly into the new `mod test` block